src/ansys/acp/core/_recursive_copy.py

```diff
@@ -0,0 +1,199 @@
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import collections
+from collections.abc import Iterable
+
+import networkx as nx
+
+from ._dependency_graph import _build_dependency_graph, _WalkTreeOptions
+from ._tree_objects import LookUpTable1D, LookUpTable1DColumn, LookUpTable3D, LookUpTable3DColumn
+from ._tree_objects._grpc_helpers.linked_object_helpers import get_linked_paths
+from ._tree_objects.base import CreatableTreeObject, TreeObject
+from ._typing_helper import StrEnum
+
+__all__ = ["recursive_copy", "LinkedObjectHandling"]
+
+
+class LinkedObjectHandling(StrEnum):
+    """Defines options for handling linked objects when copying a tree of ACP objects."""
+
+    COPY = "copy"
+    KEEP = "keep"
+    DISCARD = "discard"
+
+
+def recursive_copy(
+    *,
+    source_objects: Iterable[CreatableTreeObject],
+    parent_mapping: dict[TreeObject, TreeObject],
+    linked_object_handling: LinkedObjectHandling | str = "copy",
+) -> dict[CreatableTreeObject, CreatableTreeObject]:
+    """Recursively copy a tree of ACP objects.
+
+    This function copies a tree of ACP objects, starting from the given source objects.
+    The copied tree includes all child objects. Linked objects can optionally be included,
+    controlled by the ``linked_object_handling`` argument.
+
+    To specify where the new objects should be stored, you must provide a dictionary
+    in the ``parent_mapping`` argument. The keys of the dictionary are the original
+    parent objects, and the values are the new parent objects.
+    Note that this mapping may need to contain parent objects that are not direct parents
+    of the source objects, if another branch of the tree is included via linked objects.
+
+    The ``parent_mapping`` argument can also include objects which are part of the
+    ``source_objects`` list. In this case, the function will not create a new object for
+    the parent, but will use the existing object instead.
+
+    The function returns a list of newly created objects.
+
+    .. note::
+
+        Only attributes supported by PyACP are copied to the new objects.
+
+    Parameters
+    ----------
+    source_objects :
+        The starting point of the tree to copy.
+    parent_mapping :
+        A list of tuples defining where the new objects are stored. Each tuple contains
+        the original parent object as the first element and the new parent object as the
+        second element.
+    linked_object_handling :
```

Comment:
I am wondering if the user knows what linked objects are. I would add a minimal example.